src/backend/gporca/libnaucrates/src/md/CMDTypeBoolGPDB.cpp

```diff
@@ -296,6 +296,8 @@ CDXLDatum *
 CMDTypeBoolGPDB::GetDatumVal(CMemoryPool *mp, IDatum *datum) const
 {
 	CDatumBoolGPDB *datum_bool = dynamic_cast<CDatumBoolGPDB *>(datum);
+	GPOS_RTL_ASSERT(datum_bool);
```

Comment:
I think that here and in other similar places it will be sufficient to use GPOS_ASSERT.